src/omen/cogs/wvw_commands.py

```diff
@@ -162,6 +174,69 @@ async def get_skirmish_data(self, match_data):
             for s in self.server_colors:
                 self.match_data_dict[s]['skirmish']['skirmish_scores'][skirmish['id']] = skirmish['scores'][s]
 
+    async def predict_future_skirmish_data(self):
+        data = {
+            'skirmish_id': list(self.match_data_dict['green']['skirmish']['skirmish_scores'].keys())[:-1],
+            'green_score': list(self.match_data_dict['green']['skirmish']['skirmish_scores'].values())[:-1],
+            'blue_score': list(self.match_data_dict['blue']['skirmish']['skirmish_scores'].values())[:-1],
+            'red_score': list(self.match_data_dict['red']['skirmish']['skirmish_scores'].values())[:-1]
+        }
+        past_skirmishes = DataFrame(data,
+            index=RangeIndex(start=data['skirmish_id'][0], stop=data['skirmish_id'][-1] + 1, name="skirmish"),
+            columns=['green_score', 'blue_score', 'red_score'])
+
+        if len(past_skirmishes.index) < 2:
+            self.logger.info("not enough finished skirmishes to forecast")
+            return
+
+        # forecast the future - can mess around with the regressor and lags if needed
+        self.logger.debug(f"forecasting with {min(len(past_skirmishes.index) - 1, 4)} lags")
+        forecaster = ForecasterAutoregMultiSeries(
+            regressor = Ridge(),
+            lags = min(len(past_skirmishes.index) - 1, 4),
+            transformer_series = StandardScaler()
+        )
+        forecaster.fit(past_skirmishes)
+        remaining_skirmishes = forecaster.predict(steps=(84 - data['skirmish_id'][-1]))
+
+        # create a complete match representation based on the forecast
+        match = pandas.concat([past_skirmishes, remaining_skirmishes])
+
+        # for debugging and understanding, log the match
+        pandas.set_option('display.max_rows', None)
+        self.logger.debug(f"forecast after {data['skirmish_id'][-1]} skirmishes:")
+        self.logger.debug(f"\n{match}")
```

Comment:
Let me know if you want me to remove. I definitely want to run this locally with these so I can see the forecasts, though. I'm interested to know how good they are and when they start looking good. That'll take many matches worth of data, though, and I'm not sure how you do log management. May just comment out to make it easier to run locally and then remove later.